modin/engines/base/io/text/csv_reader.py

```diff
@@ -126,8 +129,12 @@ def _read(cls, filepath_or_buffer, **kwargs):
             # This is the number of splits for the columns
             num_splits = min(len(column_names), num_partitions)
             # This is the chunksize each partition will read
-            chunk_size = max(1, (total_bytes - f.tell()) // num_partitions)
-
+            if nrows is None:
+                chunk_size_bytes = max(1, num_partitions, total_bytes // num_partitions)
+                rows_per_part = None
+            else:
+                chunk_size_bytes = None
+                rows_per_part = max(1, num_partitions, nrows // num_partitions)
```

Comment:
You are manipulating bytes and strings at the same level of abstraction, which is not very good. Ideally, leave only working with strings, but if this does not work out, then at least separate the bytes and strings to different levels of abstraction.

Reply:
I have no success to completely get rid of using bytes and string at the same abstraction level, but I moved all that work to `TextFileReader` level to a method called `partitioned_file`, that handle partitioning file into chunks

Reply:
It seems that we don't need to separate work with bytes and strings, because we should use those values to compute base chunk size (in bytes or in rows), that further will be converted to a bytes chunk size anyway